KsApi/MockService.swift

```diff
@@ -20,14 +20,13 @@
     fileprivate let changeCurrencyResponse: GraphMutationEmptyResponseEnvelope?
     fileprivate let changeCurrencyError: GraphError?
 
-    fileprivate let changeEmailError: GraphError?
-    fileprivate let changeEmailResponse: UserEnvelope<UserEmailFields>?
+    fileprivate let changeEmailResult: Result<UpdateAccountEnvelope, ErrorEnvelope>?
```

Comment:
Given all three of these properties are returning the same envelope type, is there an opportunity to consolidate here?

Reply:
That's a good idea, and I thought it could be done at first, but we have 3 different inputs:
`CreatePasswordInput`, `ChangePasswordInput` and `ChangeEmailInput`. If we refactor those to be one generic input then yes we can consolidate all the calls here. I tried to leave them alone for now so as to reduce the overhead for this PR. Also separation for _how_ we are using those inputs matters right? Say for example we are in the `ChangePasswordViewModel` and make a service call to `UpdateUserAccount` with the `UpdateUserAccountInput`, well I lose a bit of readability as to exactly what is happening under the hood. Am I updating a password? Updating and email? Creating a password? Sometimes the separation can lead to easily following intended changes.

Reply:
Cool, just throwing it out there. Sounds like you did your due diligence on it. Going to mark this as resolved.